src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,82 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double RESIZE_FACTOR = 0.75;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        storage = (Node<K, V>[]) new Node[DEFAULT_SIZE];
+    }
 
     @Override
-    public void put(K key, V value) {
+    public int getSize() {
+        return size;
+    }
 
+    @Override
+    public void put(K key, V value) {
+        Node<K, V> existingNode = getNodeByKey(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+            return;
+        }
+        ensureCapacity();
+        insertNode(new Node<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        final Node<K, V> existingNode = getNodeByKey(key);
+        return existingNode == null ? null : existingNode.value;
     }
 
-    @Override
-    public int getSize() {
-        return 0;
+    private Node<K, V> getNodeByKey(K key) {
+        Node<K, V> current = storage[getBucketIndex(key)];
+        do {
+            if (current == null || Objects.equals(current.key, key)) {
+                return current;
+            }
+            current = current.next;
+        } while (true);
```

Comment:
To write infinite loop in project is not so good idea. 
You can rewrite it using simple while: `while(current != null)`

Reply:
https://github.com/mate-academy/jv-hash-map/pull/860#discussion_r1118609762
I used while(true) to get rid of unnecessary return statement as requested in the above comment.
replacing while(true) with while(current != null) will cause compilation error with requirement to add return null; after while statement.

Reply:
Moreover, current == null will always trigger break from the cycle in 'if (current == null...' block